### ISSUE TITLE
Allow expression in transforms

### ISSUE THREAD

**Issue by max-sixty** [PRQL/prql#812]
So, for example, this would work:

```elm
from employees
sort (salary + benefits)
```

This won't be easy to implement, but hopefully could be quite general and allow for lots of extensibility without adding much complication to the language. (Until then, it require an additional `derive` to create a variable).

@aljazerzen will be out for a while, and he's probably the best person to build it; I can try and have a go too. It's probably not an ideal first PR!

**Comment by mklopets** [comment]
I think this can now be closed, post-0.3.0

The example PRQL now compiles to:
```sql
WITH table_1 AS (
  SELECT
    *,
    salary + benefits AS _expr_0
  FROM
    employees
  ORDER BY
    _expr_0
)
SELECT
  *
FROM
  table_1
```

**Comment by max-sixty** [comment]
Yes!!
